src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt

```diff
@@ -93,6 +100,69 @@ class DownloadTaskIT : KafkaTestContainersUtils() {
         Assertions.assertThat(resultFile2.readText()).contains("description")
     }
 
+    @ParameterizedTest(name = "{0}")
+    @ArgumentsSource(SchemaArgumentProvider::class)
+    fun `Should format supported schema types when pretty is specified`(type: String, oldSchema: ParsedSchema,
+                                                                        newSchema: ParsedSchema) {
+        // Given
+        val subjectName = "parameterized-$type"
+
+        client.register(subjectName, oldSchema)
+        client.register(subjectName, newSchema)
+
+        buildFile = folderRule.newFile("build.gradle")
+        buildFile.writeText(
+            """
+            plugins {
+                id 'java'
+                id 'com.github.imflog.kafka-schema-registry-gradle-plugin'
+            }
+
+            schemaRegistry {
+                url = '$schemaRegistryEndpoint'
+                pretty = true
+                download {
+                    subject('$subjectName', '${folderRule.root.absolutePath}/src/main/$type/test')
+                    subject('$subjectName', 'src/main/$type/test_v1', 1)
+                    subject('$subjectName', 'src/main/$type/test_v2', 2)
+                }
+            }
+        """
+        )
+
+        // When
+        val result: BuildResult? = GradleRunner.create()
+            .withGradleVersion("7.6")
+            .withProjectDir(folderRule.root)
+            .withArguments(DownloadTask.TASK_NAME)
+            .withPluginClasspath()
+            .withDebug(true)
+            .build()
+
+        // Then
+        val schemaType = newSchema.schemaType().toSchemaType()
+
+        val schemaFile = "$subjectName.${schemaType.extension}"
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test")).exists()
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test/$schemaFile")).exists()
+        Assertions.assertThat(result?.task(":downloadSchemasTask")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+        if (schemaType != SchemaType.PROTOBUF) {
+            Assertions.assertThat(File(folderRule.root, "src/main/$type/test/$schemaFile")).hasContent(
+                objectMapper.readTree(newSchema.toString()).toPrettyString()
+            )
+        }
+
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test_v1")).exists()
+        val resultFile1 = File(folderRule.root, "src/main/$type/test_v1/$schemaFile")
+        Assertions.assertThat(resultFile1).exists()
+        Assertions.assertThat(resultFile1.readText()).doesNotContain("description")
+
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test_v2")).exists()
+        val resultFile2 = File(folderRule.root, "src/main/$type/test_v2/$schemaFile")
+        Assertions.assertThat(resultFile2).exists()
+        Assertions.assertThat(resultFile2.readText()).contains("description")
```

Comment:
Meybe to simplify this test and only check the formatting you could remove the "new schema" and only register one schema and verify that it's retrieved correctly ?